File: src/main/java/com/jocoos/mybeautip/domain/file/code/FileType.java
package com.jocoos.mybeautip.domain.file.code;

import com.jocoos.mybeautip.global.code.CodeValue;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum FileType implements CodeValue {

    IMAGE("이미지"),
    VIDEO("비디오");

    private final String description;

    @Override
    public String getName() {
        return name();
    }
}

File: src/main/kotlin/com/mybeautip/global/code/FileType.kt
package com.mybeautip.global.code

import com.jocoos.mybeautip.global.code.CodeValue

enum class FileType(val desc: String) : CodeValue {

    IMAGE("이미지 파일"),
    VIDEO("동영상 파일");

    override fun getDescription(): String = this.desc

    override fun getName(): String {
        return this.name
    }
}